src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py

```diff
@@ -375,13 +419,27 @@ def get_suggestion(self, random_search=False):
         total_params : dict
             parameter suggestion
         """
+        if self.parallel and len(self.total_data)>20 and len(self.running_data):
+            self.CL_rval = copy.deepcopy(self.rval)
+            _constant_liar_y = 0
+            if self.constant_liar_type == 'mean' and self.optimal_y[1]:
```

Comment:
could you make sure self.optimal_y must be an array here? and if self.optimal_y[1] is 0, assigning self.optimal_y to _constant_liar_y is not correct.

Reply:
I think in the original design :
when constant_liar_type == 'mean' --> self.optimal_y is a array
otherwise --> self.optimal is a value.

Reply:
I added two more comments which are the reason that I have concern here.

Reply:
discussed with @PurityFan , to fix this problem, you can simply add `self.optimal_y is not None` in line 422

Reply:
can remove `self.optimal_y[1]` from line 424

Reply:
@QuanluZhang I don't think we could remove sel.optimal_y[1] in line 424 because it will avoid /0 in line 425.

Reply:
@QuanluZhang  Other comments fix in new PR.

Reply:
@xuehui1991 if self.optimal_y is not None, self.optimal_y[1] must be larger than 0.

Reply:
@QuanluZhang  changed. Please review new PR.